core/src/tile/tileID.h

```diff
@@ -41,14 +44,30 @@ struct TileID {
         return isValid() && z <= _maxZoom;
     }
 
+    TileID withMaxSourceZoom(int32_t _max) const {
```

Comment:
I think this name makes usage at the call site pretty clear, e.g. 
```c++
requestTileId = tileId.withMaxSourceZoom(source.maxZoom());
```
"unzoomed" seems misleading since tiles below the max zoom are unchanged by the call.